app/src/components/pages/collectionsPage/collectionsPage.tsx

```diff
@@ -181,16 +126,21 @@ export function CollectionsPage({ data, params, renderCollections }) {
           id={"search-collections"}
           textInputProps={{
             isClearable: true,
-            isClearableCallback: () => {
-              setcurrentCollectionKeywords(DEFAULT_SEARCH_TERM);
-            },
+            isClearableCallback: () =>
+              collectionSearchManager.handleKeywordChange(DEFAULT_SEARCH_TERM),
             labelText: "Search by collection title",
             name: "collection_keywords",
             placeholder: "Search by collection title",
-            defaultValue: currentCollectionKeywords,
-            onChange: (e) => handleSearchChange(e),
+            defaultValue: collectionSearchManager.keywords,
+            onChange: (e) =>
+              collectionSearchManager.handleKeywordChange(
+                (e.target as HTMLInputElement).value
+              ),
+          }}
+          onSubmit={() => {
+            setIsLoaded(false);
+            updateURL(collectionSearchManager.handleSearchSubmit());
```

Comment:
Can you try adding `updateURL` inside of `setIsLoaded` as part of the callback function? I don't see a difference but just trying to reduce async issues.